doc/user/methods/adaboost_regressor.md

```diff
@@ -0,0 +1,582 @@
+   representing the `i`th decision tree in the adaboost regressor.
+
+For complete functionality, the [source
+code](/src/mlpack/methods/adaboost/adaboost_regressor.hpp) can be consulted.
+Each method is fully documented.
+
+### Simple Examples
+
+See also the [simple usage example](#simple-usage-example) for a trivial use of 
+`AdaBoostRegressor`.
+
+---
+
+Train a adaboost regressor on mixed categorical data.
+
+```c++
+  // Load a categorical dataset.
+arma::mat data;
+mlpack::data::DatasetInfo info;
+// See https://datasets.mlpack.org/telecom_churn.arff.
+mlpack::data::Load("telecom_churn.arff", data, info, true);
+
+arma::rowvec responses;
+// See https://datasets.mlpack.org/telecom_churn.responses.csv.
+mlpack::data::Load("telecom_churn.responses.csv", responses, true);
+
+// Split data into training set (80%) and test set (20%).
+arma::mat trainData, testData;
+arma::rowvec trainResponses, testResponses;
+mlpack::data::Split(data, responses, trainData, testData, trainResponses,
+    testResponses, 0.2);
+
+// Create the tree.
+mlpack::AdaBoostRegressor<> adr;
+// Train on the given dataset, specifying a minimum gain of 1e-7 and keeping the
+// default minimum leaf size.
+const double mse = adr.Train(trainData, info, trainResponses, 20 /*number of trees*/
+    ,10 /* minimum leaf size */, 1e-7 /* minimum gain */, 4 /*maximum depth*/);
+// Print the MSE of the trained tree.
+std::cout << "MSE of trained tree is " << mse << "." << std::endl;
+
+// Compute prediction on the first test point.
+const double firstPrediction = adr.Predict(testData.col(0));
+std::cout << "Predicted value for first test point is " << firstPrediction
+    << "." << std::endl;
+
+// Compute predictions on test data.
+arma::rowvec testPredictions;
+adr.Predict(testData, testPredictions);
+
+// Compute the average error on the test set.
+const double testAverageError = arma::mean(testResponses - testPredictions);
+std::cout << "Average error on test set: " << testAverageError << "."
+    << std::endl;
+```
+
+---
+
+Load a adaboost regressor and print some information about it.
+
+```c++
+mlpack::AdaBoostRegressor<> adr;
+// This call assumes a adaboost regressor called "adr" has already been saved to
+// `adr.bin` with `data::Save()`.
+mlpack::data::Load("adr.bin", "adr", adr, true);
+
+std::cout << "The adaboost regressor in 'adr.bin' contains " << adr.NumTrees()
+    << " trees." << std::endl;
+if (adr.NumTrees() > 0)
+{
+  std::cout << "The first tree's root node has " << adr.Tree(0).NumChildren()
+      << " children." << std::endl;
+}
+```
+
+---
+
+Train a adaboost regressor on categorical data, and compare its performance 
+with the performance of each individual tree:
+
+```c++
+// Load a categorical dataset.
+arma::mat data;
+mlpack::data::DatasetInfo info;
+// See https://datasets.mlpack.org/telecom_churn.arff.
+mlpack::data::Load("telecom_churn.arff", data, info, true);
```

Comment:
Just for the sake of variety of examples, it might be useful to use a different dataset here.  In other examples I've tried to use a handful of datasets throughout.